platforms/gke/base/core/container_cluster/container_cluster.tf

```diff
@@ -267,10 +218,57 @@ resource "google_container_cluster" "cluster" {
   }
 }
 
-data "google_container_cluster" "default" {
-  depends_on = [google_container_cluster.cluster]
+resource "google_container_node_pool" "system" {
+  # Variables
+  cluster            = google_container_cluster.cluster.name
+  initial_node_count = 1
+  location           = var.cluster_region
+  name               = "system"
+  project            = google_container_cluster.cluster.project
+
+  # Blocks
+  autoscaling {
+    location_policy      = "BALANCED"
+    total_max_node_count = 1000
+    total_min_node_count = 1
+  }
+
+  network_config {
+    enable_private_nodes = true
+  }
 
-  location = var.cluster_region
-  name     = local.cluster_name
-  project  = var.cluster_project_id
+  node_config {
+    # Variables
+    labels = {
+      "resource-type" : "system"
+    }
+    machine_type    = var.cluster_system_node_pool_machine_type
+    service_account = google_service_account.cluster.email
```

Comment:
Is it possible to have a variable here and use `google_service_account.cluster.email` is the user didn't specify anything? This will be useful for the federated learning use case